File: testdata/next.golden/api/documents/route.go
package documents

import (
	"net/http"

	"github.com/emersonjoe/trilha"
)

// /api/documents, ported from app/api/documents/route.ts.
//
// Source: 15 lines, server component.
func GET(c *trilha.Ctx) error {
	return trilha.Errorf(http.StatusNotImplemented, "GET /api/documents has not been ported yet")
}

func POST(c *trilha.Ctx) error {
	return trilha.Errorf(http.StatusNotImplemented, "POST /api/documents has not been ported yet")
}

func DELETE(c *trilha.Ctx) error {
	return trilha.Errorf(http.StatusNotImplemented, "DELETE /api/documents has not been ported yet")
}

File: testdata/next/app/api/documents/route.ts
import { NextResponse } from "next/server"

export async function GET(request: Request) {
  return NextResponse.json({ documents: [] })
}

export async function POST(request: Request) {
  const body = await request.json()
  return NextResponse.json(body, { status: 201 })
}

export async function DELETE(request: Request) {
  return new NextResponse(null, { status: 204 })
}
